spec/forms/openid_connect_authorize_form_spec.rb

```diff
@@ -0,0 +1,128 @@
+require 'rails_helper'
+
+RSpec.describe OpenidConnectAuthorizeForm do
+  subject(:form) do
+    OpenidConnectAuthorizeForm.new(
+      acr_values: acr_values,
+      client_id: client_id,
+      nonce: nonce,
+      prompt: prompt,
+      redirect_uri: redirect_uri,
+      response_type: response_type,
+      scope: scope,
+      state: state
+    )
+  end
+
+  let(:acr_values) do
+    'http://idmanagement.gov/ns/assurance/loa/1 http://idmanagement.gov/ns/assurance/loa/3'
+  end
+
+  let(:client_id) { 'urn:gov:gsa:openidconnect:test' }
+  let(:nonce) { SecureRandom.hex }
+  let(:prompt) { 'select_account' }
+  let(:redirect_uri) { 'gov.gsa.openidconnect.test://result' }
+  let(:response_type) { 'code' }
+  let(:scope) { 'openid profile' }
+  let(:state) { SecureRandom.hex }
+
+  describe '#valid?' do
```

Comment:
Sorry, I misspoke. `valid?` is a public method because we are including `ActiveModel::Model`. I thought that we weren't calling `#valid?` outside of the class, but I just noticed we are in the controller `index` action. However, we are also calling `#submit` in the `create` action, so we need to test `#submit` as well.